monitor/hosts.go

```diff
@@ -116,29 +116,30 @@ func terminateHosts(ctx context.Context, hosts []host.Host, settings *evergreen.
 			for hostToTerminate := range work {
 				event.LogMonitorOperation(hostToTerminate.Id, reason)
 
-				err := util.RunFunctionWithTimeout(func() error {
-					return terminateHost(ctx, hostToTerminate, settings)
-				}, 12*time.Minute)
-
-				if err != nil {
-					if strings.Contains(err.Error(), ec2.EC2ErrorNotFound) {
-						err = hostToTerminate.Terminate()
-						if err != nil {
-							catcher.Add(errors.Wrap(err, "unable to set host as terminated"))
-							continue
+				func() { // use a function so that the defer works
+					var cancel context.CancelFunc
+					ctx, cancel = context.WithTimeout(ctx, 3*time.Minute)
+					defer cancel()
+
+					if err := terminateHost(ctx, hostToTerminate, settings); err != nil {
+						if strings.Contains(err.Error(), ec2.EC2ErrorNotFound) {
+							err = hostToTerminate.Terminate()
+							if err != nil {
+								catcher.Add(errors.Wrap(err, "unable to set host as terminated"))
+								return
+							}
+							grip.Debugf("host %s not found in EC2, changed to terminated", hostToTerminate.Id)
+							return
 						}
-						grip.Debugf("host %s not found in EC2, changed to terminated", hostToTerminate.Id)
-						continue
-					}
-					if err == util.ErrTimedOut {
-						catcher.Add(errors.Errorf("timeout terminating host %s", hostToTerminate.Id))
-						continue
+						if err == util.ErrTimedOut {
+							catcher.Add(errors.Errorf("timeout terminating host %s", hostToTerminate.Id))
+							return
+						}
+						err = errors.Wrapf(err, "error terminating host %s", hostToTerminate.Id)
+						catcher.Add(err)
+						grip.Warning(err)
```

Comment:
Is it necessary to log and also add to the catcher? This will end up getting logged at an error level anyway

Reply:
done